velox/vector/fuzzer/VectorFuzzer.cpp

```diff
@@ -88,13 +98,49 @@ int64_t rand(FuzzerGenerator& rng) {
   return boost::random::uniform_int_distribution<int64_t>()(rng);
 }
 
+template <typename T>
+typename std::enable_if<std::is_floating_point_v<T>, T>::type
+generateNumericLimits(FuzzerGenerator& rng) {
+  if (coinToss(rng, 0.25)) {
+    return std::numeric_limits<T>::quiet_NaN();
+  }
+
+  if (coinToss(rng, 0.25)) {
+    return -std::numeric_limits<T>::quiet_NaN();
+  }
+
+  if (coinToss(rng, 0.25)) {
+    return std::numeric_limits<T>::infinity();
+  }
+
+  return -std::numeric_limits<T>::infinity();
+}
+
 template <>
 double rand(FuzzerGenerator& rng) {
+  if (coinToss(rng, FLAGS_velox_fuzzer_nan_inf_threshold)) {
+    return generateNumericLimits<double>(rng);
+  }
+
+  // Generate a -tive number with 50% probability.
+  if (coinToss(rng, 0.5)) {
+    return -boost::random::uniform_01<double>()(rng);
+  }
+
   return boost::random::uniform_01<double>()(rng);
 }
 
 template <>
 float rand(FuzzerGenerator& rng) {
+  if (coinToss(rng, FLAGS_velox_fuzzer_nan_inf_threshold)) {
```

Comment:
This version for float can be combined with the version for double. Something like
```
template <typename T>
typename std::enable_if_t<std::is_floating_point_v<T>, T> rand(FuzzerGenerator& rng) {
  if (coinToss(rng, FLAGS_velox_fuzzer_nan_inf_threshold)) {
    return generateNumericLimits<T>(rng);
  }
  
  // Generate a -tive number with 50% probability.
  if (coinToss(rng, 0.5)) {
    return -boost::random::uniform_01<T>()(rng);
  }

  return boost::random::uniform_01<T>()(rng);
}
```